fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java

```diff
@@ -0,0 +1,121 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.iceberg;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableSet;
+import com.starrocks.connector.PartitionUtil;
+import org.apache.iceberg.AddedRowsScanTask;
+import org.apache.iceberg.ChangelogOperation;
+import org.apache.iceberg.ChangelogScanTask;
+import org.apache.iceberg.DeletedDataFileScanTask;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.IncrementalChangelogScan;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IcebergPartitionUtils {
+    private static final Logger LOG = LogManager.getLogger(IcebergPartitionUtils.class);
+    public static class IcebergPartition {
+        private PartitionSpec spec;
+        private StructLike data;
+        private ChangelogOperation operation;
+
+        IcebergPartition(PartitionSpec spec, StructLike data, ChangelogOperation operation) {
+            this.spec = spec;
+            this.data = data;
+            this.operation = operation;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof IcebergPartition)) {
+                return false;
+            }
+            IcebergPartition that = (IcebergPartition) o;
+            return Objects.equal(spec, that.spec) &&
+                    Objects.equal(data, that.data) && operation == that.operation;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(spec, data, operation);
+        }
+    }
+
+    public static Set<String> getChangedPartitionNames(Table table, long from) {
+        Set<IcebergPartition> changedPartition = getChangedPartition(table, from);
+        return changedPartition.stream().map(partition -> PartitionUtil.
+                convertIcebergPartitionToPartitionName(table.spec(), partition.data)).collect(Collectors.toSet());
+    }
+
+    public static Set<IcebergPartition> getChangedPartition(Table table, long from) {
+        ImmutableSet.Builder<IcebergPartition> builder = ImmutableSet.builder();
+        Snapshot snapShot = table.currentSnapshot();
+        if (snapShot.timestampMillis() >= from) {
+            while (snapShot.parentId() != null) {
+                snapShot = table.snapshot(snapShot.parentId());
+                if (snapShot.timestampMillis() <= from) {
+                    break;
+                }
+            }
+            if (snapShot.timestampMillis() <= from) {
```

Comment:
why <=

Reply:
find the snapshot which timestamp is **from** or snapshot which timestamp is before **from**